### ISSUE TITLE
visualizing uncertainty graphs

### ISSUE THREAD

**Issue by 61966** [Ephemeral182/UDR-S2Former_deraining#3]
This is a great job, but can the author provide code for visualizing uncertainty graphs so that the work can be better disseminated.

**Comment by Ephemeral182** [comment]
Thanks for your attention to our job. It is an easy thing to visualize uncertainty graphs, which can be seen by only using the heat map of the matplotlib package, as follows: 
def get_cmap(x: torch.Tensor, path: str):
    import matplotlib.pyplot as plt
    x_ = x.squeeze(0).cpu().numpy()
    x_mean = x_.transpose(1,2,0).mean(2)
    plt.imshow(x_mean,cmap='summer_r')
    plt.savefig(path)
where x denotes the feature level of the uncertainty map. For example, you can obtain the final uncertainty map via nn.Sequential(*[nn.Conv2d(dim[0],dim[0],3,1,1), nn.ELU(),nn.Conv2d(dim[0],out_cahnnels,3,1,1)]), just like the line 839 in the UDR_S2Former.py.

**Comment by x33rak** [comment]
I am getting undesired uncertainty maps when trying to replicate the uncertainty maps from your paper. Would you happen to know why this may be occurring? 

![image](https://github.com/user-attachments/assets/db2c4517-3752-4066-9033-1f3bb8e1fc3c)
